consumers/sender/main.go

```diff
@@ -0,0 +1,230 @@
+		return nil
+	}
+
+	count, err := h.s.CountLogsByUUID(msg.UUID)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"user_id":     msg.UserID,
+			"campaign_id": msg.CampaignID,
+			"uuid":        msg.UUID,
+		}).WithError(err).Error("Unable to count sent logs")
+		return nil
+	}
+
+	if count > 0 {
+		logrus.WithFields(logrus.Fields{
+			"user_id":     msg.UserID,
+			"campaign_id": msg.CampaignID,
+			"uuid":        msg.UUID,
+		}).Warn("Email already sent.")
+		return nil
+	}
+
+	input := &ses.SendEmailInput{
+		Destination: &ses.Destination{
+			ToAddresses: []*string{aws.String(msg.SubscriberEmail)},
+		},
+		Message: &ses.Message{
+			Body: &ses.Body{
+				Html: &ses.Content{
+					Charset: aws.String(CharSet),
+					Data:    aws.String(string(msg.HTMLPart)),
+				},
+				Text: &ses.Content{
+					Charset: aws.String(CharSet),
+					Data:    aws.String(string(msg.TextPart)),
+				},
+			},
+			Subject: &ses.Content{
+				Charset: aws.String(CharSet),
+				Data:    aws.String(string(msg.SubjectPart)),
+			},
+		},
+		Source: aws.String(msg.Source),
+		Tags: []*ses.MessageTag{
+			{
+				Name:  aws.String("campaign_id"),
+				Value: aws.String(strconv.FormatInt(msg.CampaignID, 10)),
+			},
+			{
+				Name:  aws.String("user_id"),
+				Value: aws.String(msg.UserUUID),
+			},
+		},
+	}
+
+	if msg.ConfigurationSetExists {
+		input.ConfigurationSetName = aws.String(emails.ConfigurationSetName)
+	}
+
+	resp, err := client.SendEmail(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ses.ErrCodeMessageRejected:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Message rejected.")
+			case ses.ErrCodeMailFromDomainNotVerifiedException:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Domain not verified.")
+			case ses.ErrCodeConfigurationSetDoesNotExistException:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Configuration set does not exist.")
+			}
+		}
+		logrus.WithFields(logrus.Fields{
+			"user_id":     msg.UserID,
+			"campaign_id": msg.CampaignID,
+			"uuid":        msg.UUID,
+		}).WithError(err).Error("Unable to send bulk templated email. Unknown status code.")
+		return nil
+	}
+
+	err = h.s.CreateSendLog(&entities.SendLog{
+		UUID:         msg.UUID,
+		UserID:       msg.UserID,
+		CampaignID:   msg.CampaignID,
+		SubscriberID: msg.SubscriberID,
+		Status:       "successful",
```

Comment:
Move this to constants. Also, I'm thinking about whether or not we should store some things in the send log from the `SendEmail` response. Can we take a look at the response?

Reply:
![image](https://user-images.githubusercontent.com/35736870/106372940-28194400-6375-11eb-9eb1-33e52eb92c9b.png)

Reply:
I'm thinking that maybe we should set the `message_id` in the send log, so we can compare it with the hooks in the future. It's worth having it there.